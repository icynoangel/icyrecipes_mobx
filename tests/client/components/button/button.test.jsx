import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import Button from './../../../../src/client/js/components/button/button';
import enzymeToJson from 'enzyme-to-json';

describe('<Button />', function() {
  this.getComponent = type => {
    return <Button onClick={this.onClick} caption={this.caption} type={type} />;
  };

  beforeEach(() => {
    this.onClick = jest.fn();
    this.caption = 'Get Items';
  });

  it('Should render a primary Button', () => {
    const wrapper = shallow(this.getComponent('primary'));
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render a secondary Button', () => {
    const wrapper = shallow(this.getComponent('secondary'));
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should call onClick function', () => {
    const wrapper = shallow(this.getComponent('secondary'));
    wrapper.find('button').simulate('click');
    expect(this.onClick).toHaveBeenCalledTimes(1);
  });
});
