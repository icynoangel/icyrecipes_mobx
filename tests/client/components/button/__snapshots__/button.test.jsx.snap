// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render a primary Button 1`] = `
<button
  className="button --primary"
  onClick={[MockFunction]}
>
  Get Items
</button>
`;

exports[`<Button /> Should render a secondary Button 1`] = `
<button
  className="button --secondary"
  onClick={[MockFunction]}
>
  Get Items
</button>
`;
